queue/QueueConsumer.js

```diff
@@ -1,16 +1,44 @@
 const BaseQueue = require('./BaseQueue')
 const generateTagName = require('./util/generateTagName')
+const assertQueue = require('./util/assertQueue')
+
+const DEFAULT_PREFETCH_COUNT = 10
+const DEFAULT_MAX_MESSAGE_RETRIES = 5
+const DEAD_LETTER_HEADER = 'x-death'
+
+const AMQ_FANOUT = 'fanout'
 
 module.exports = class QueueConsumer extends BaseQueue {
-  constructor ({queueName, connection, prefetchCount, logger}) {
+  constructor (options) {
+    const {
+      queueName,
+      connection,
+      prefetchCount,
+      logger,
+      maxMessageRetries,
+      deadLetterQueueName
+    } = options
+
     super({
       queueName,
       connection,
       tag: generateTagName(queueName, 'consumer'),
       logger
     })
 
-    this._prefetchCount = prefetchCount || 10
+    this._prefetchCount = prefetchCount || DEFAULT_PREFETCH_COUNT
+    this._deadLetterExchangeName = `${queueName}-dead-letter-exchange`
+
+    this._deadLetterQueueName = deadLetterQueueName ||
+      `${queueName}-dead-letter-queue`
+
+    this._maxMessageRetries = isNaN(maxMessageRetries)
```

Comment:
This is pretty unlikely, but if someone passes `null` for `options.maxMessageRetries`, `isNaN` will be `false` and `maxMessageRetries` will be `0`. Passing `undefined` to `isNaN` results in `true` so it would be okay in the typical case.